code/game/atoms.dm

```diff
@@ -81,6 +81,10 @@
 		reagents = null
 	invisibility = 101
 	return ..()
+	
+//Hook for running code when a dir change occurs
+/atom/proc/setDir(newdir)

```

Comment:
Or, we can _not_ rename a proc, and just mass-replace everything contained in here with `set_dir` instead of `setDir` 😛